gluten-substrait/src/main/scala/org/apache/gluten/expression/UnaryExpressionTransformer.scala

```diff
@@ -43,7 +44,10 @@ case class CastTransformer(substraitExprName: String, child: ExpressionTransform
   extends UnaryExpressionTransformer {
   override def doTransform(args: java.lang.Object): ExpressionNode = {
     val typeNode = ConverterUtils.getTypeNode(dataType, original.nullable)
-    ExpressionBuilder.makeCast(typeNode, child.doTransform(args), original.ansiEnabled)
+    ExpressionBuilder.makeCast(
+      typeNode,
+      child.doTransform(args),
+      SparkShimLoader.getSparkShims.withAnsiEvalMode(original))
```

Comment:
Yes, I note Spark also sets `ansiEnabled = true` for `EvalMode.TRY`, which is for re-using the code logic of `EvalMode.ANSI`. The difference is, in TRY mode any exception is caught and then NULL is returned.
Velox has different implementation, which requires us to set this flag simply according to whether EvalMode is ANSI or not.

@acvictor, please leave some comments here to clarify, which should be helpful for future code maintenance.

https://github.com/apache/spark/blob/v3.5.3/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala#L471